variables.tf

```diff
@@ -125,3 +125,25 @@ variable "transit_encryption_mode" {
   type        = string
   default     = "SERVER_AUTHENTICATION"
 }
+
+variable "maintenance_policy" {
+  description = "The maintenance policy for an instance."
+  type = object({
+    day = string
+    start_time = object({
+      hours   = number
+      minutes = number
+      seconds = number
+      nanos   = number
+    })
+  })
+  default = {
+    day = "SATURDAY"
+    start_time = {
+      hours   = 4
+      minutes = 30
+      seconds = 0
+      nanos   = 0
+    }
+  }
```

Comment:
Let's set the default to empty to maintain backward compatibility and use api default.

Reply:
@bharathkkb maybe I'm missing something, but to define the variable type as such:
```
  type = object({
    day = string
    start_time = object({
      hours   = number
      minutes = number
      seconds = number
      nanos   = number
    })
  })
```
will require the default value to be set with something other than: `default: {}`